code/Player/Roles/RoleHandler.cs

```diff
@@ -64,16 +64,20 @@ public void AssignRandomRole()
         /// <returns></returns>
         public Role GetRandomRoleFromPreferences()
         {
-            int totalPoints = rolePreferences.Sum(rolePreference => rolesFactors[rolePreference.Value]);
-            System.Random rnd = new();
+            int totalPoints = 0;
+            foreach ((Role role, RolePreference value) in rolePreferences)
+            {
+                totalPoints += rolesFactors[value];
+            }
+            Random rnd = new();
```

Comment:
I think we need to give a seed to a Random object

Reply:
No need. when you don't give seed its random. 
But maybe we can make a server SEED generator and you can start game with seed :o